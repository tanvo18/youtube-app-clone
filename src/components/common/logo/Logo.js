import React from 'react';
import LogoStyled from './LogoStyled';
import PropTypes from 'prop-types';

const Logo = ({ logoClass, logoImg, link, logoWidth, logoHeight, logoAlt }) => {
  return (
    <LogoStyled
      className={logoClass}
      logoWidth={logoWidth}
      logoHeight={logoHeight}
    >
      <a href={link}>
        <img src={logoImg} alt={logoAlt} />
      </a>
    </LogoStyled>
  );
};

Logo.defaultProps = {
  logoImg: '',
  link: '',
  logoWidth: '',
  logoHeight: '',
  logoAlt: '',
};

Logo.propType = {
  logoImg: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  logoWidth: PropTypes.string,
  logoHeight: PropTypes.string,
  logoAlt: PropTypes.string,
};

export default Logo;