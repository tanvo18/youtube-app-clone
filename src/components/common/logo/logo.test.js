import React from 'react';
import Logo from './Logo';
import renderer from 'react-test-renderer';

const props = {
  logoWidth: '10px',
  logoHeight: '10px',
  logoImg: '../../assets/images/youtube.png',
  logoAlt: 'this is a logo',
  link: '/'
};

describe('test logo', () => {
  it('should render a logo', () => {
    const wrapper = shallow(<Logo {...props} />);
    expect(wrapper.find('img').exists()).toBe(true);
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<Logo {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});