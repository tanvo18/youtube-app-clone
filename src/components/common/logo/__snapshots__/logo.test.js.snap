// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test logo should render correctly 1`] = `
<h1
  className="sc-bdVaJa kHzMix"
>
  <a
    href="/"
  >
    <img
      alt="this is a logo"
      src="../../assets/images/youtube.png"
    />
  </a>
</h1>
`;
